src/main/java/org/sasanlabs/service/vulnerability/csrf/CSRFVulnerability.java

```diff
@@ -0,0 +1,249 @@
+            @RequestParam(USR_PASS) String usrPass,
+            @RequestParam(USR_NEWPASS) String usrNewPass,
+            @RequestParam(TOKEN_PARAM) String usrLiveToken)
+            throws IOException {
+        Map<String, List<String>> headers = new HashMap<>();
+        headers.put(
+                "Set-Cookie", Arrays.asList("CSRF_Level_4=" + FIXED_TOKEN + "; HttpOnly; Secure"));
+        return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                new GenericVulnerabilityResponseBean<String>(
+                        checkInputsToken(usrLiveToken, FIXED_TOKEN, usrId, usrPass, usrNewPass),
+                        true),
+                CollectionUtils.toMultiValueMap(headers),
+                HttpStatus.OK);
+    }
+
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.CSRF,
+            description = "CSRF_VULNERABILITY_USER_INPUT1",
+            payload = "CSRF_PAYLOAD_LEVEL_5")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_5,
+            htmlTemplate = "LEVEL_5/CSRFVulnerability",
+            variant = Variant.SECURE)
+    public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerableCSRFLvl5(
+            @RequestParam(USR_LOGIN) String usrId,
+            @RequestParam(USR_PASS) String usrPass,
+            @RequestParam(USR_NEWPASS) String usrNewPass,
+            @RequestParam(TOKEN_PARAM) String usrLiveToken)
+            throws IOException {
+        Map<String, List<String>> headers = new HashMap<>();
+        if (usrLiveToken.equals("secretrequest")) {
+            secretToken = Integer.toString(rand.nextInt(1000000000));
+            headers.put(
+                    "Set-Cookie",
+                    Arrays.asList("CSRF_Level_5_secret=" + secretToken + "; HttpOnly; Secure"));
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(secretToken, true),
+                    CollectionUtils.toMultiValueMap(headers),
+                    HttpStatus.OK);
+        } else {
+            headers.put(
+                    "Set-Cookie",
+                    Arrays.asList("CSRF_Level_5_secret=" + FIXED_TOKEN + "; HttpOnly; Secure"));
+            return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
+                    new GenericVulnerabilityResponseBean<String>(
+                            checkInputsToken(usrLiveToken, secretToken, usrId, usrPass, usrNewPass),
+                            true),
+                    HttpStatus.OK);
+        }
+    }
+
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.CSRF,
+            description = "CSRF_VULNERABILITY_USER_INPUT2",
+            payload = "CSRF_PAYLOAD_LEVEL_5")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_6,
+            htmlTemplate = "LEVEL_6/CSRFVulnerability",
+            variant = Variant.SECURE,
```

Comment:
this is not a secure level, i would suggest going through https://www.youtube.com/watch?v=hW2ONyxAySY&ab_channel=TroyHunt video for information about CSRF.

Reply:
and https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html for mitigating this risk in applications.